### ISSUE TITLE
Custom record URL

### ISSUE THREAD

**Issue by JackYoustra** [NextLevel/NextLevel#132]
Hello all,
   Recording in a session doesn't support custom targets. This is an issue: if a shot is interrupted halfway through recording, I can't export it to its intended destination if I have a custom filestructure / naming convention. Is there any way around this?

**Comment by piemonte** [comment]
hey @JackYoustra. you have access to each clip's URL even when an interruption occurs. then move it to the desired location.

NextLevel.shared.session?.clips?.first.url

https://github.com/NextLevel/NextLevel/blob/1917286065d4760856076967ca3225d7794e4e28/Sources/NextLevelClip.swift#L46-L50

**Comment by JackYoustra** [comment]
Yes, but if the user runs out of battery, the video is unrecoverable - the file is in a temporary directory and there's no handler that is guarunteed to be call in a low-power environment.

**Comment by piemonte** [comment]
ah, i understand the scenario now. it'd be possible to add support for specifying a set of urls before capture but it'd require the paths to be validated and unique.

**Comment by JackYoustra** [comment]
This is an easy to meet requirement for the use case in mind

**Comment by Goktug** [comment]
Any update?
